### PR TITLE
[3D] notify user if the 3D scene export failed

### PR DESCRIPTION
**Pull request by ptitjano**
## Description

A message is now displayed to notify that the export failed.

### CODE REVIEW

File: src/3d/qgs3dmapscene.cpp [hunk 1]
```diff
@@ -1045,7 +1045,12 @@ void Qgs3DMapScene::exportScene( const Qgs3DMapExportSettings &exportSettings )
   if ( mTerrain )
     exporter.parseTerrain( mTerrain, "Terrain" );
 
-  exporter.save( exportSettings.sceneName(), exportSettings.sceneFolderPath() );
+  const bool sceneSaved = exporter.save( exportSettings.sceneName(), exportSettings.sceneFolderPath() );
+  if ( !sceneSaved )
+  {
+    QgsMessageOutput::showMessage( tr( "3D exporter warning" ), tr( "The scene can not be exported." ), QgsMessageOutput::MessageText );
```
Comment (nyalldawson) [inline src/3d/qgs3dmapscene.cpp:1051]: Can we use a messagebar message here instead? (It'd be nice to show a "success" message too, just like we do eg here: https://github.com/qgis/QGIS/blob/master/src/app/layout/qgslayoutdesignerdialog.cpp#L2268 )

Comment (ptitjano) [reply src/3d/qgs3dmapscene.cpp]: Done.

File: src/app/3d/qgs3dmapcanvaswidget.cpp [hunk 2]
```diff
@@ -565,10 +565,21 @@ void Qgs3DMapCanvasWidget::exportScene()
...
+  layout->addWidget( &exportWidget, 1 );
   layout->addWidget( buttons );
   if ( dlg.exec() )
-    w.exportScene();
+  {
+    const bool success = exportWidget.exportScene();
+    const QString exportFilePath = QDir( exportSettings.sceneFolderPath() ).filePath( exportSettings.sceneName() + QStringLiteral( ".obj" ) );
+    if ( success )
+    {
+      mMessageBar->pushMessage( tr( "Export 3D scene" ), tr( "Successfully exported scene to <a href=\"%1\">%2</a>" ).arg( exportFilePath, QDir::toNativeSeparators( exportFilePath ) ), Qgis::MessageLevel::Success, 0 );
```
Comment (nyalldawson) [inline src/app/3d/qgs3dmapcanvaswidget.cpp:576]:
```suggestion
      mMessageBar->pushMessage( tr( "Export 3D scene" ), tr( "Successfully exported scene to <a href=\"%1\">%2</a>" ).arg( QUrl::fromLocalFile( exportFilePath ), QDir::toNativeSeparators( exportFilePath ) ), Qgis::MessageLevel::Success, 0 );
```

File: src/app/3d/qgs3dmapcanvaswidget.cpp [hunk 3]
```diff
@@ -565,10 +565,21 @@ void Qgs3DMapCanvasWidget::exportScene()
...
+    }
+    else
+    {
+      mMessageBar->pushMessage( tr( "Export 3D scene" ), tr( "Unable to export scene to <a href=\"%1\">%2</a>" ).arg( exportFilePath, QDir::toNativeSeparators( exportFilePath ) ), Qgis::MessageLevel::Warning, 0 );
```
Comment (nyalldawson) [inline src/app/3d/qgs3dmapcanvaswidget.cpp:580]:
```suggestion
      mMessageBar->pushMessage( tr( "Export 3D scene" ), tr( "Unable to export scene to <a href=\"%1\">%2</a>" ).arg( QUrl::fromLocalFile( exportFilePath ), QDir::toNativeSeparators( exportFilePath ) ), Qgis::MessageLevel::Warning, 0 );
```

**Review by nyalldawson — APPROVED** [review]

**Comment by ptitjano** [comment]
Fixed the issue introduced from the applied suggestion. A `toString()` call was missing.

**Review by lbartoletti — APPROVED** [review]

**Pull request merged**